### PR TITLE
Architecture/20 - Implement Complex object Init / Destructor

### PR DESCRIPTION
**Pull request by LucasTesnier**
Implementing Init and Destroy for the following object :
- Channel
- Comment
- Conversation
- Message
- Team
- Thread
- User

Adding testing policy.

### CODE REVIEW

**Review by Cavonstavant — CHANGES_REQUESTED** [review]
Minor refactoring needed everything's ok otherwise

File: objects/comment/include/object_comment.h [hunk 1]
```diff
@@ -37,11 +42,27 @@ typedef struct comment_s {
...
     /// author uuid for the comment creation
-    my_uuid_t author;
+    my_uuid_t *author;
 } comment_creation_t;
 
+///
+/// \brief Init a comment object
+///
+/// \param content content of the comment object
+/// \return comment_t* Newly created content object
```
Comment (osvegn) [inline objects/comment/include/object_comment.h:56]: Comment instead of content

File: objects/conversation/src/object_conversation.c [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+conversation_t *conversation_init(conversation_creation_t content)
+{
+    conversation_t *new_conv = malloc(sizeof(conversation_t) * 1);
+
+    if (new_conv == CONVERSATION_ERROR)
+        return CONVERSATION_ERROR;
+    if ((new_conv->uuid = my_uuid_init(CON_)) == UUID_ERROR)
+        return CONVERSATION_ERROR;
+    if ((new_conv->participant = malloc(sizeof(my_uuid_t *) * 10))
```
Comment (osvegn) [inline objects/conversation/src/object_conversation.c:21]: Why 10 ? Use define

File: objects/conversation/src/object_conversation.c [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+        == UUID_ERROR)
+        return CONVERSATION_ERROR;
+    if ((new_conv->messages = malloc(sizeof(my_uuid_t *) * 250)) == UUID_ERROR)
```
Comment (osvegn) [inline objects/conversation/src/object_conversation.c:24]: Why 250 ? Use define

File: objects/message/src/object_message.c [hunk 4]
```diff
@@ -7,4 +7,28 @@
...
+#include "object_message.h"
+#include <stdlib.h>
+
+message_t *message_init(message_creation_t content)
+{
+    message_t *new_message = malloc(sizeof(message_t) * 1);
+
+    if (new_message == MESSAGE_ERROR)
+        return MESSAGE_ERROR;
+    if ((new_message->uuid = my_uuid_init(DMS_)) == UUID_ERROR)
```
Comment (osvegn) [inline objects/message/src/object_message.c:19]: Dirty

File: objects/message/src/object_message.c [hunk 5]
```diff
@@ -7,4 +7,28 @@
...
+        return MESSAGE_ERROR;
+    if ((new_message->body = body_init(content.body)) == BODY_ERROR)
```
Comment (osvegn) [inline objects/message/src/object_message.c:21]: Dirty

File: objects/team/src/object_team.c [hunk 6]
```diff
@@ -7,4 +7,39 @@
...
+#include "object_team.h"
+#include <stdlib.h>
+
+team_t *team_init(team_creation_t content)
+{
+    team_t *new_team = malloc(sizeof(team_t) * 1);
+
+    if (new_team == TEAM_ERROR)
+        return TEAM_ERROR;
+    if ((new_team->name = name_init(content.name)) == NAME_ERROR)
```
Comment (osvegn) [inline objects/team/src/object_team.c:19]: Dirty

File: objects/team/src/object_team.c [hunk 7]
```diff
@@ -7,4 +7,39 @@
...
+        return TEAM_ERROR;
+    if ((new_team->description = description_init(content.description))
```
Comment (osvegn) [inline objects/team/src/object_team.c:21]: Dirty

File: objects/team/src/object_team.c [hunk 8]
```diff
@@ -7,4 +7,39 @@
...
+        == DESCRIPTION_ERROR)
+        return TEAM_ERROR;
+    if ((new_team->uuid = my_uuid_init(TEM_)) == UUID_ERROR)
+        return TEAM_ERROR;
+    if ((new_team->channels = malloc(sizeof(my_uuid_t *) * 20)) == UUID_ERROR)
```
Comment (osvegn) [inline objects/team/src/object_team.c:26]: Why 20 ? Use define

File: objects/team/src/object_team.c [hunk 9]
```diff
@@ -7,4 +7,39 @@
...
+        return TEAM_ERROR;
+    if ((new_team->users = malloc(sizeof(my_uuid_t *) * 40)) == UUID_ERROR)
```
Comment (osvegn) [inline objects/team/src/object_team.c:28]: Why 40 ? Use define

**Review by osvegn — CHANGES_REQUESTED** [review]
In documentation, for the param balise, note the type of the param, instead of the name of the variable.
Stop assign value in conditional, its dirty.
Note that if a pull request can be split in differents part, do it, it will be better for reviewers.

File: objects/message/src/object_message.c (hunk 4 above)
Comment (Cavonstavant) [reply objects/message/src/object_message.c:19]: no it's not

File: objects/conversation/src/object_conversation.c (hunk 2 above)
Comment (LucasTesnier) [reply objects/conversation/src/object_conversation.c:21]: Temporary, will be modified in the issue associated with the object.

File: objects/conversation/src/object_conversation.c (hunk 3 above)
Comment (LucasTesnier) [reply objects/conversation/src/object_conversation.c:24]: Temporary, will be modified in the issue associated with the object.

File: objects/team/src/object_team.c (hunk 8 above)
Comment (LucasTesnier) [reply objects/team/src/object_team.c:26]: Temporary, will be modified in the issue associated with the object.

File: objects/team/src/object_team.c (hunk 9 above)
Comment (LucasTesnier) [reply objects/team/src/object_team.c:28]: Temporary, will be modified in the issue associated with the object.

**Pull request merged**